yew/src/services/websocket.rs

```diff
@@ -172,13 +180,21 @@ impl WebSocketService {
         &mut self,
         url: &str,
         notification: &Callback<WebSocketStatus>,
-    ) -> Result<ConnectCommon, &str> {
+    ) -> Result<ConnectCommon, WebSocketError> {
         let ws = WebSocket::new(url);
-        if ws.is_err() {
-            return Err("Failed to created websocket with given URL");
-        }
+        #[cfg(feature="web_sys")]
+        if let Err(ws_error) = &ws {
+            return Err(WebSocketError::CreationError(ws_error.clone().unchecked_into::<js_sys::Error>().to_string().as_string().unwrap()));
```

Comment:
I'd prefer not having an `unwrap()` here. Is there a more reliable way to handle web-sys errors?

Reply:
I'm not sure. I took inspiration for this from the fetch service but I'll investigate.